src/azul/service/responseobjects/manifest_service.py

```diff
@@ -0,0 +1,60 @@
+import base64
+import json
+
+from azul import config
+from azul.service.responseobjects.step_function_helper import StateMachineError, StepFunctionHelper
+
+
+class ManifestService:
+    """
+    Class containing logic for starting and checking the status of manifest generation jobs
+    """
+    manifest_endpoint = '/repository/manifest/files'
+
+    step_function_helper = StepFunctionHelper()
+
+    def encode_params(self, params, encoding='utf-8'):
+        return base64.urlsafe_b64encode(bytes(json.dumps(params), encoding=encoding)).decode(encoding)
+
+    def decode_params(self, token, encoding='utf-8'):
+        return json.loads(base64.urlsafe_b64decode(token).decode(encoding))
+
+    def start_manifest_generation(self, filters):
+        """
+        Start the execution of a state machine generating the manifest
+
+        :param filters: filters to use for the manifest
+        :return: The id of the execution of the state machine
+        """
+        return self.step_function_helper.start_execution(
+            config.manifest_state_machine_name,
+            execution_input={'filters': filters}
+        )['executionArn'].split(':')[-1]
+
+    def get_manifest_status(self, params, wait, local=False):
```

Comment:
I'm uncomfortable with how the logic of parsing request input and generating the response output is spread over three layers: two functions in `app.py` and this method. They all seem to be tightly coupled. The fact that you have to extract the `manifest_endpoint` and use it in the route in app.py is the bad smell here.

I also understand that we don't want `app.py` to grow without bound. I do want all routes to be defined in a single `app.py` module so I think the way forward is to have the functions in `app.py` only extract the URL path and request parameters but do everything else in the service layer.

Reply:
The reason I made two functions in `app.py` is so I can directly call the second function from a unit test without having to do `requests.get`.  We should be able to move most of what's in `app.py` into `manifest_service.py` and just have one function in each file